### ISSUE TITLE
[Feature] Add validation set support in SIMBA and GEPA

### ISSUE THREAD

**Issue by JovanVeljanoski** [stanfordnlp/dspy#9062]
### What feature would you like to see?

It would be very helpful if both SIMBA and GEPA supported the use of a validation dataset. So as one run finishes, the best proposal is also evaluated on a separate validation dataset (which is not used for the training/evolution process). 

I believe this is already supported in the MIPRO_V2 optimizer

Thanks!

### Would you like to contribute?

- [ ] Yes, I'd like to help implement this.
- [x] No, I just want to request it.

**Comment by chenmoneygithub** [comment]
@JovanVeljanoski  Thanks for the feature request! 

GEPA already has this valset in `compile()` method, which is not used for evolution, but only for validation purposes. 

SIMBA works a bit differently that it only utilizes a trainset, which is the nature of the algorithm. We may change that in the future, but to now that's by intention.

**Comment by JovanVeljanoski** [comment]
Hi, 

Thanks for the quick response. I noticed that GEPA has support for validation - thanks! 

Thanks for the clarification on SIMBA. I suppose I can do this semi-manually :). 
So far I am getting the best results with SIMBA on all of my tasks pretty much, so was looking to do more rigorous tests with less effort :)

Thanks again!

(p.s. and kudos to the team, this library is quite promising, I am becoming a fan! Keep up the great work, and thank you!)

**Comment by chenmoneygithub** [comment]
@JovanVeljanoski  Thank you for using DSPy!
